Index: rtl/objpas/sysutils/sysint.inc
===================================================================
--- rtl/objpas/sysutils/sysint.inc	(revision 35644)
+++ rtl/objpas/sysutils/sysint.inc	(working copy)
@@ -14,6 +14,50 @@
  **********************************************************************}
 
 
+{ ---------------------------------------------------------------------
+    TFormatSettings
+  ---------------------------------------------------------------------}
+
+
+Class Function TFormatSettings.Create: TFormatSettings;
+
+begin
+  result:=DefaultFormatSettings;
+end;
+
+Class Function TFormatSettings.Invariant: TFormatSettings;
+
+Const
+  INVARIANT_FORMAT_SETTINGS: TFormatSettings = (
+    CurrencyFormat: 0;
+    NegCurrFormat: 0;
+    ThousandSeparator: ',';
+    DecimalSeparator: '.';
+    CurrencyDecimals: 2;
+    DateSeparator: '/';
+    TimeSeparator: ':';
+    ListSeparator: ',';
+    CurrencyString: #$a4;
+    ShortDateFormat: 'MM/dd/yyyy';
+    LongDateFormat: 'dddd, dd MMMMM yyyy HH:mm:ss';
+    TimeAMString: 'AM';
+    TimePMString: 'PM';
+    ShortTimeFormat: 'HH:mm';
+    LongTimeFormat: 'HH:mm:ss';
+    ShortMonthNames: ('Jan','Feb','Mar','Apr','May','Jun', 
+                      'Jul','Aug','Sep','Oct','Nov','Dec');
+    LongMonthNames: ('January','February','March','April','May','June',
+                     'July','August','September','October','November','December');
+    ShortDayNames: ('Sun','Mon','Tue','Wed','Thu','Fri','Sat');
+    LongDayNames:  ('Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday');
+    TwoDigitYearCenturyWindow: 50;
+  );
+
+begin
+  result:=INVARIANT_FORMAT_SETTINGS;
+end;
+
+
 procedure InitInternationalGeneric;
   begin
     fillchar(SysLocale,sizeof(SysLocale),0);
Index: rtl/objpas/sysutils/sysinth.inc
===================================================================
--- rtl/objpas/sysutils/sysinth.inc	(revision 35644)
+++ rtl/objpas/sysutils/sysinth.inc	(working copy)
@@ -31,6 +31,7 @@
   TWeekNameArray = array[1..7] of string;
 
   TFormatSettings = record
+  public
     CurrencyFormat: Byte;
     NegCurrFormat: Byte;
     ThousandSeparator: Char;
@@ -51,6 +52,8 @@
     ShortDayNames: TWeekNameArray;
     LongDayNames: TWeekNameArray;
     TwoDigitYearCenturyWindow: Word;
+    class function Create: TFormatSettings; static; {$ifdef SYSTEMINLINE}inline;{$endif}
+    class function Invariant: TFormatSettings; static;
   end;
 
 var
